### DISCUSSION TITLE
When attempting to run the boids example I get unknown type: `arrayLength`

### DISCUSSION THREAD

**Discussion by 13ros27** [gfx-rs/wgpu#1488] [Q&A] [answered]
I downloaded the boids example including framework.rs and tried to run it but I get:
```
Using Intel(R) HD Graphics (Dx12)
[2021-06-10T17:57:57Z ERROR wgpu_core::device] Failed to parse WGSL code: unknown type: `arrayLength`
wgpu error: Validation Error

Caused by:
    In Device::create_shader_module
    Failed to parse WGSL
```
It seems to not know what arrayLength is from compute.wgsl however I don't know enough about wgpu to have any idea what is causing it?
For dependencies I used:
```
async-executor = "*"
bytemuck = "*"
cgmath = "*"
env_logger = "*"
log = "*"
pollster = "*"
rand = "*"
wgpu = "*"
winit = "*"
```

**Comment by kvark** [comment, marked as answer]
Hi!
The example code is written against the implementation code, which is ahead of the last crates publish.
Please see https://github.com/gfx-rs/wgpu-rs/tree/v0.8/examples for the relevant example code for v0.8.

**Reply by 13ros27** [reply]
Thank you, changing the wgsl file for that one fixed it
